File: AgentHistorique/backend/app/schemas/monument.py
from __future__ import annotations

from pydantic import BaseModel, Field


class MonumentSummaryResponse(BaseModel):
    id: float = Field(..., description="Monument identifier (may be fractional)")
    name_fr: str
    name_en: str | None = None
    name_ar: str | None = None
    latitude: float
    longitude: float
    visit_duration_min: float | None = None
    dominant_period: str | None = None
    function: str | None = None
    popularity: float | None = None
    image_url: str | None = None


class MonumentsListResponse(BaseModel):
    monuments: list[MonumentSummaryResponse]

File: frontend/src/types/monument.ts
export interface MonumentSummary {
  id: number;
  name_fr: string;
  name_en: string | null;
  name_ar: string | null;
  latitude: number;
  longitude: number;
  visit_duration_min: number | null;
  dominant_period: string | null;
  function: string | null;
  popularity: number | null;
  image_url: string | null;
}

export interface MonumentsListResponse {
  monuments: MonumentSummary[];
}

export function monumentDisplayName(
  monument: MonumentSummary,
  language: "fr" | "en" | "ar" = "fr",
): string {
  if (language === "en" && monument.name_en) return monument.name_en;
  if (language === "ar" && monument.name_ar) return monument.name_ar;
  return monument.name_fr;
}
